rsocket/internal/Allowance.h

```diff
@@ -8,23 +8,23 @@
 
 namespace rsocket {
 
-class AllowanceSemaphore {
+class Allowance {
  public:
   using ValueType = size_t;
 
-  AllowanceSemaphore() = default;
+  Allowance() = default;
 
-  explicit AllowanceSemaphore(ValueType initialValue) : value_(initialValue) {}
+  explicit Allowance(ValueType initialValue) : value_(initialValue) {}
 
-  bool tryAcquire(ValueType n = 1) {
-    if (!canAcquire(n)) {
+  bool tryConsume(ValueType n) {
```

Comment:
Do we ever really want to do `tryConsume(n)` where `n > 1`?  Seems like we don't other than tests...

I'd figure we *have* to always consume by 1, since we're calling `onNext()` one-by-one, and any of them can cancel the subscription at any given time.

Reply:
If that's the case, maybe this method should be `consumeOne()` and tests fixed to expect only one allowance consumed. I feel like at least this is more explicit about what's going on, rather than assuming a default value of 1.